sdk/python/kfp/components_tests/test_components.py

```diff
@@ -720,6 +720,43 @@ def test_prevent_passing_unserializable_objects_as_argument(self):
         with self.assertRaises(TypeError):
             component(input_1="value 1", input_2=open)
 
+    def test_input_output_uri_resolving(self):
+        component_text = textwrap.dedent('''\
+            inputs:
+            - {name: In1}
+            outputs:
+            - {name: Out1}
+            implementation:
+              container:
+                image: busybox
+                command:
+                - program
+                - --in1-uri
+                - {inputUri: {inputName: In1, supportedSchemes: [GoogleCloudStorage]}}
```

Comment:
For my education: is this only an annotation or we do have some schema check during compile time now?

Reply:
If something like this info is part of the component interface (e.g. the input type spec), then it should be checked by the SDK when composing or compiling a pipeline.
However when this info is part of the implementation, the information should not affect the pipeline author, but instead be intended for the execution engine, so that it can launch this container component implementation.

If you have `supportedSchemes: [GoogleCloudStorage]`, then the system must give you a GCS URI.
If you have `supportedSchemes: [AmazonS3]`, then the system must give you an S3 URI.

It's the execution system responsibility to make sure it can give the requested kind of URI. For example, if you pass an Artifact that has artifact.uri=https://.... to `supportedSchemes: [GoogleCloudStorage]`, the system must provide the `gs://...` URI for that data to the component code.
This does not seem easy to do. Which is one of the reasons the `ContainerSpec` does not support the URI placeholders.